src/gui/mzroll/tabledockwidget.cpp

```diff
@@ -1103,10 +1103,33 @@ void TableDockWidget::deleteGroups() {
             if (parentGroup->deleteChild(group)) {
                 QTreeWidgetItem *parentItem = item->parent();
                 if (parentItem) {
-                	parentItem->removeChild(item);
-              		delete (item);
-            	}
-          	}
+                    parentItem->removeChild(item);
+                    delete (item);
+
+                    // once a child is deleted, the pointers storing the
+                    // location of memory blocks of child `PeakGroup` objects,
+                    // may no longer be valid, therefore we update them.
+                    for (int i = 0; i < parentItem->childCount(); ++i) {
```

Comment:
@saifulbkhan might be simpler to clear all child items and start afresh. What do you think?

Reply:
I had thought of this, but wasn't sure how that would impact user experience when an isotopic child was deleted from a parent having many other children. Deleting and repopulating 5+ child nodes might lead to blocky behavior. I could still give it a try.

Reply:
I just tried out what you suggested. For a group having 9 isotopic children, deleting one made the UI freeze for 2 seconds while clearing & re-entering was in progress. So, can we stick to the current logic?

Reply:
Current logic is fine. Was just trying to simplify this re:maintenance.

I'm only concerned about the behavior if user clicks on any of the children while the data is being updated. Will try and test for it as much as possible.